src/mapping/PartitionOfUnityMapping.hpp

```diff
@@ -0,0 +1,385 @@
+
+  /// tag the vertices required for the mapping
+  virtual void tagMeshFirstRound() override;
+
+  /// nothing to do here
+  virtual void tagMeshSecondRound() override;
+
+private:
+  precice::logging::Logger _log{"mapping::PartitionOfUnityMapping"};
+
+  std::vector<Partition<RADIAL_BASIS_FUNCTION_T>> _partitions;
+
+  // Shape parameter or support radius for the RBF interpolant,
+  // only required for the Partition instantiation
+  const double _parameter;
+
+  // Input parameter
+  const unsigned int _verticesPerPartition;
+  const double       _relativeOverlap;
+  const bool         _projectToInput;
+
+  // Derived parameter
+  double averagePartitionRadius = 0;
+
+  /// true if the mapping along some axis should be ignored
+  /// has currently only dim x false entries, as integrated polynomials are irrelevant
+  std::vector<bool> _deadAxis;
+
+  Polynomial _polynomial;
+
+  // Holds the output vertex -> partition association. Outer vector has the size of the output mesh and inner vector size of the associated partitions
+  std::vector<std::vector<VertexID>> _partMap;
+
+  /// @copydoc Mapping::mapConservative
+  virtual void mapConservative(DataID inputDataID, DataID outputDataID) override;
+
+  /// @copydoc Mapping::mapConsistent
+  virtual void mapConsistent(DataID inputDataID, DataID outputDataID) override;
+
+  void exportPartitionCentersAsVTU(mesh::Mesh &centers);
+};
+
+// --------------------------------------------------- HEADER IMPLEMENTATIONS
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+PartitionOfUnityMapping<RADIAL_BASIS_FUNCTION_T>::PartitionOfUnityMapping(
+    Mapping::Constraint constraint,
+    int                 dimension,
+    double              parameter,
+    Polynomial          polynomial,
+    unsigned int        verticesPerPartition,
+    double              relativeOverlap,
+    bool                projectToInput)
+    : Mapping(constraint, dimension),
+      _parameter(parameter), _verticesPerPartition(verticesPerPartition), _relativeOverlap(relativeOverlap), _projectToInput(projectToInput), _polynomial(polynomial)
+{
+  PRECICE_ASSERT(this->getDimensions() <= 3);
+  PRECICE_CHECK(_relativeOverlap < 1, "The relative overlap has to be smaller than one.");
+  PRECICE_CHECK(_verticesPerPartition > 0, "The number of vertices per partition has to be greater zero.");
+
+  if (isScaledConsistent()) {
+    setInputRequirement(Mapping::MeshRequirement::FULL);
+    setOutputRequirement(Mapping::MeshRequirement::FULL);
+  } else {
+    setInputRequirement(Mapping::MeshRequirement::VERTEX);
+    setOutputRequirement(Mapping::MeshRequirement::VERTEX);
+  }
+  // Simply set each axis to be active according to the space dimension.
+  // We could even set each axis to be true, which would just lead to 'zero' columns in the polynomial QR.
+  // Integrated polynomials are anyway not supported with PoU
+  _deadAxis.clear();
+  std::fill_n(std::back_inserter(_deadAxis), getDimensions(), false);
```

Comment:
I am not sure yet if I agree, but anyway: we should handle this in the configuration.